File: Swift/1_Two_Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numIndices = [Int: Int]()
        for (index, num) in nums.enumerated() {
            let complement = target - num
            if let complementIndex = numIndices[complement] {
                return [complementIndex, index]
            }
            numIndices[num] = index
        }
        return []
    }
}

File: Rust/1_Two_Sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut num_indices = std::collections::HashMap::new();

        for (i, &num) in nums.iter().enumerate() {
            let complement = target - num;

            if let Some(&index) = num_indices.get(&complement) {
                return vec![index as i32, i as i32];
            }

            num_indices.insert(num, i);
        }

        vec![]
    }
}